rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,424 @@
+
+"""RDKit Conformer Browser
+Derived from Greg Landrum's code by Malitha Humayun Kabir
+As a part of GSoC 2017
+Project : RDKit - 3Dmol.js integration
+Mentors: Paul Czodrowski and Greg Landrum
+Date: 17th July 2017
+Email# malitha12345@gmail.com
+"""
+
+import py3Dmol
+from rdkit import Chem
+from rdkit.Chem import AllChem
+from rdkit.Chem import Descriptors
+from ipywidgets import Layout, Label, Button, Box, HBox, VBox
+from ipywidgets import Dropdown, SelectMultiple, IntSlider, HTML, Checkbox, Button
+from IPython.display import display
+import time
+
+
+bgcolors_3d = ['0xeeeeee', '0x000000', '0xffffff']
+
+prop_rdkit=sorted([prop for prop,obj in Descriptors._descList])
+
+drawing_types_3d=['line', 'cross', 'stick', 'cartoon', 'sphere', 'surface', 'ballstick']
+
+color_scheme_3d=['default', 'greenCarbon', 'cyanCarbon', 'magentaCarbon', 
+                 'yellowCarbon', 'whiteCarbon', 'orangeCarbon', 'purpleCarbon', 
+                 'blueCarbon', 'ssPyMOL', 'ssJmol', 'Jmol', 'amino', 
+                 'shapely', 'nucleic', 'chain', 'chainHetatm', 'prop']
+
+
+def ProcessMolContainingObj(Mol):
+    """This function checks whether the object type fits the requirements for rendering.
+    If the oject doesn't have necessary attributes, it takes action to include that.
+    """
+    try:
+        moldict = dict(Mol)
+    except TypeError:
+        if type(Mol) is tuple:
+            moldict=list()
+            moldict.append(Mol)
+            moldict = dict(moldict)
+        elif hasattr(Mol, '__iter__') is False:
+            moldict=list()
+            moldict.append(('m0', Mol))
+            moldict = dict(moldict)
+        elif type(Mol) is list:
+            Mol_keys=['m'+str(x) for x in range(len(Mol))]
+            Mol_2=[(Mol_keys[i],Mol[i]) for i in range(len(Mol))]
+            moldict = dict(Mol_2)
+    return moldict
```

Comment:
I agree user should know if any function returns `None`. This has been followed too.